specification/trace/semantic_conventions/http.md

```diff
@@ -153,6 +153,13 @@ This span type represents an outbound HTTP request. There are two ways this can
 
 For an HTTP client span, `SpanKind` MUST be `Client`.
 
+HTTP client spans SHOULD end sometime after all the response headers are received. This MAY or MAY NOT include time spent reading the response body.
+
+HTTP client library instrumentations SHOULD document their behavior around ending HTTP client spans.
+
+> **Note**
+> It is nice for HTTP client spans to end once the response body has been fully read. However, if there is any possibility for application code to not fully read the HTTP response (and for the HTTP client library to then have to clean up the HTTP response asynchronously), the HTTP client span SHOULD be ended earlier (e.g. after the HTTP response headers are read). This avoids having to end the span asynchronously later on at a time which is no longer directly associated with the application code which made the HTTP request.
```

Comment:
Update the description to mention that HTTP requests end with either reading the response body, or being closed with an error.
```suggestion
> It is nice for HTTP client spans to represent the complete lifetime of the HTTP request. An HTTP request ends once the response body has been fully read or the response closes with an error. However, if there is any possibility for application code to not fully read the HTTP response (and for the HTTP client library to then have to clean up the HTTP response asynchronously), the HTTP client span SHOULD be ended earlier (e.g. after the HTTP response headers are read). This avoids having to end the span asynchronously later on at a time which is no longer directly associated with the application code which made the HTTP request.
```